methods/__init__.py

```diff
@@ -0,0 +1,9 @@
+from abc import ABC, abstractmethod
+
+from utils.result import ResultType
+
+
+class DetectionMethod(ABC):
+    @abstractmethod
+    def run(self) -> ResultType:
```

Comment:
arent there params missing here?

Reply:
Yes and no, each method could have a different set of parameters. We could set it to the current one used by the dst, but most probably we would need to change it in the future.

Reply:
Aight just keep it for now then